bin/ci-publish.sh

```diff
@@ -2,35 +2,44 @@
 
 set -ex pipefail
 
-npmPublish() {
-  npm publish
-}
-
-currentVersion() {
+getCurrentVersion() {
   grep version <package.json | sed 's/[^0-9.]*\([0-9.]*\).*/\1/' | head -n 1
 }
 
-currentProject() {
+getCurrentProject() {
   npm ls --depth=-1 | head -1 | cut -f 1 -d " " | sed 's/@.*//'
 }
 
-checkVersion() {
-  local gitTag
-  gitTag=$(git describe --tags)
+getLatestTag() {
+  git describe --abbrev=0 --tags
+}
+
+getCommitHash() {
+  git rev-list -n 1 $1
+}
+
+checkHeadIsLatestTag() {
+  if [ "$(getCommitHash HEAD)" != "$(getCommitHash $(getLatestTag))" ]; then
```

Comment:
Yeah, without `abbrev=0` it's not the actual tag name, which is why the previous code was confusing